lib/rouge/lexers/haskell.rb

```diff
@@ -146,21 +146,8 @@ def self.detect?(text)
       end
 
       state :character do
-        rule /\\/ do
-          token Str::Escape
-          push :character_end
```

Comment:
this should have been a `goto` as well - it doesn't matter much since you've deleted it, but that's part of what was causing the bug.

Reply:
It sounds better! Thanks.

2be2622